diff --git a/backend/svirt.pm b/backend/svirt.pm
index 2402311..5d66f8f 100644
--- a/backend/svirt.pm
+++ b/backend/svirt.pm
@@ -49,6 +49,7 @@ sub do_start_vm {
         {
             hostname => get_required_var('VIRSH_HOSTNAME'),
             password => get_var('VIRSH_PASSWORD'),
+            username => get_var('VIRSH_USERNAME') || 'root',
         });
 
     $ssh->backend($self);
@@ -79,7 +80,7 @@ sub run_cmd {
     $self->{ssh} = $self->new_ssh_connection(
         hostname => get_required_var('VIRSH_HOSTNAME'),
         password => get_var('VIRSH_PASSWORD'),
-        username => 'root'
+        username => get_var('VIRSH_USERNAME') || 'root',
     );
     my $chan = $self->{ssh}->channel();
     $chan->exec($cmd);
@@ -135,7 +136,7 @@ sub load_snapshot {
 sub start_serial_grab {
     my ($self, $name) = @_;
 
-    my $chan = $self->start_ssh_serial(hostname => get_required_var('VIRSH_HOSTNAME'), password => get_var('VIRSH_PASSWORD'), username => 'root');
+    my $chan = $self->start_ssh_serial(hostname => get_required_var('VIRSH_HOSTNAME'), password => get_var('VIRSH_PASSWORD'), username => (get_var('VIRSH_USERNAME') || 'root'));
     if (check_var('VIRSH_VMM_FAMILY', 'vmware')) {
         # libvirt esx driver does not support `virsh console', so
         # we have to connect to VM's serial port via TCP which is
diff --git a/consoles/sshVirtsh.pm b/consoles/sshVirtsh.pm
index b3a9e6d..1005e57 100644
--- a/consoles/sshVirtsh.pm
+++ b/consoles/sshVirtsh.pm
@@ -50,8 +50,9 @@ sub activate {
 
     my $hostname = $args->{hostname} || die('we need a hostname to ssh to');
     my $password = $args->{password};
+    my $username = $args->{username};
 
-    $self->{ssh} = $self->backend->new_ssh_connection(hostname => $hostname, password => $password);
+    $self->{ssh} = $self->backend->new_ssh_connection(hostname => $hostname, password => $password, username => $username);
     if ($self->vmm_family eq 'vmware') {
         $self->{sshVMwareServer} = $self->backend->new_ssh_connection(
             hostname => get_required_var('VMWARE_SERVER'),
